Add a CTA on /earn/payments-plans to upgrade

dsas opened the pull request:
#### Changes proposed in this Pull Request

People without a WordPress.com subscription can access
/earn/payments-plans if they have previously connected their site to
Stripe (see #62616). They can only view and delete their plans though,
not add and edit them. In order to add or edit their plans they need to
have a current WordPress.com subscription.

This change adds an UpgradeNudge (with related stats tracking) to make that clear, and to invite them to upgrade.



#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Be on a paid plan
2. Go to wordpress.com/earn (or Tools -> Earn) in the menu
3. Follow the prompts to link your stripe account as normal
4. Click on one of the stripe-related features on /earn
5. Click on Payment plans
6. Notice you can add plans and edit them (via the ellipses menu)
7. Cancel your paid plan
8. Click on one of the stripe-related features on /earn
9. Open your browser dev console and enter `localStorage.setItem('debug', 'calypso:analytics*');`
10. Click on Payment plans
11. Notice that there is a nudge to upgrade plans
13. Click the nudge, it should take you to the /plans page to upgrade
14. Check your browser console, you should see three events:
   * Bump stats event for clicking the nudge - `calypso_earn_page:payment-plans-upgrade-button`
   * Tracks event for viewing the nudge - `calypso_earn_page_payment_plans_upgrade_button_view`
   * Tracks event for clicking the nudge - `calypso_earn_page_payment_plans_upgrade_button_click`
   * You should also be able to see the two Tracks events in Tracks Live by searching for your user name.

Before | After
-------|------
![image](https://user-images.githubusercontent.com/93301/162756454-9427f738-bf89-4054-85ca-32a7ed2f0d0b.png)  | ![image](https://user-images.githubusercontent.com/93301/165349058-36b1b7aa-2f72-4ae5-a7f9-0d574554a559.png)


<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to suggestion made in #62616

dsas committed:
Add a CTA on /earn/payments-plans to upgrade

People without a WordPress.com subscription can access
/earn/payments-plans if they have previously connected their site to
Stripe (see #62616). They can only view and delete their plans though,
not add and edit them. In order to add or edit their plans they need to
have a WordPress.com subscription.

This change adds a button to make that clear.

dsas commented on client/my-sites/earn/memberships/products.jsx:
```diff
@@ -74,13 +74,18 @@ class MembershipsProductsSection extends Component {
-				) }
+					) }
+					{ ! this.props.hasStripeFeature && (
+						<Button primary compact href={ '/plans/' + this.props.siteSlug }>
```
Presumably this needs some tracks events adding

dsas replied:
There's some existing tracking on the /earn page:
 * The stats for plan upgrades on /earn are used on the mission control /s/calypso_earn_page stats, this is powered by bump stats.
 * There is also the `calypso_earn_page_upgrade_button_click` Tracks stat (properties are plan: wp.com plan or any-paid-plan, feature:donations/recurring-payments/etc).

 The /earn/payments-plans page is used for managing plans for donations, premium-content, and manage payments, it doesn't actually know (or care) which of those features was used to get to the page. 

If reusing the `calypso_earn_page_upgrade_button_click` event name, it could use 'any-feature' as the event name analogous to any-paid-plan, though the wording in Tracks all talks about the cards on the /earn page, so perhaps this isn't a good idea. I'm leaning towards coining a new tracks name.

Is the bump stats tracking needed here at all? It feels like it probably shouldn't reuse the same name if so - it's not an exactly equal comparison with the existing values.

Copons replied:
> I'm leaning towards coining a new tracks name.

I agree.
I'd probably mimic the path with a `calypso_earn_page_payment_plans_upgrade_button_click`, with no properties.

And I'll also keep the bump as well, reusing the original `calypso_earn_page` event, with a `payment-plans-upgrade-button` feature for easy comparison.

dsas committed:
Add tracking

dsas commented:
Editorial people:

I have added a button that reads 'Upgrade to add end edit plans', I'm looking for any suggestions or improvements on that, or sign-off if it's ok.

The plans being referred to are payment plans managed by a site administrator to use on their website for subscribers, donors or purchasers of (usually non-physical) goods to send them money.

On the root screen for this flow - /earn, we have buttons on cards titled 'Unlock this feature', so that may be better alternative wording

See https://wordpress.com/support/wordpress-editor/blocks/payments/#payments-management for further background and context

dsas commented:
> Is there a reason why we didn't went for an upgrade nudge design like we have for themes? It would be consistent with what we have right now

Interesting point! There's an argument to be made that the new upgrade CTA is in the same place as the primary _Add new plan_ button is for people who do have a plan. So people _might_ be expecting to see a button there anyway (not that _Add new plan_ is a frequent action), but I think you are probably right.

dsas commented:
Moved back to In progress so I can follow up with @BogdanUngureanu's feedback on the upgrade nudge.

Copons commented:
Casually pinging @SaxonF and @vinimotaa for design input!

dsas committed:
Payment plans: Refactor upgrade to UpsellNudge

Rather than a button in place of the 'Add new plan' button, use an
UpsellNudge instead.

dsas committed:
Payment plans: Put SectionHeader inside if

dsas commented:
**Editorial folk**:
I have added an UpsellNudge that reads 'Upgrade to change payment plans or add new plans', I'm looking for any suggestions or improvements on that.

The plans being referred to are payment plans managed by a site administrator to use on their website for subscribers, donors or purchasers of (usually non-physical) goods to send them money.

The person viewing the nudge needs to upgrade their WordPress.com plan so that they have access to add/edit the plans that their visitors can use to pay the site owner.

On the root screen for this flow - /earn, we have buttons on cards titled 'Unlock this feature', so that may be better alternative wording

See https://wordpress.com/support/wordpress-editor/blocks/payments/#payments-management for further background and context

![image](https://user-images.githubusercontent.com/93301/165348216-4062cfdf-e715-4449-9dfb-2094fcf96d15.png)

kristastevens commented:
Hello from Editorial @dsas, 

I think what you have is clear and straightforward. I prefer it over phrases that use "unlock." Unlock feels a bit fake and gimmicky. One alternative wording suggestion for you: 

Original:

> Upgrade to change payment plans or add new plans

Suggestion: 

> Upgrade to modify payment plans or add new plans

dsas committed:
Use improved wording from editorial

dsas commented:
Thanks @kristastevens

SaxonF commented:
Looks good @dsas !

BogdanUngureanu reviewed (approved):
LGTM! :shipit:

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7283009

Thank you @dsas for including a screenshot in the description! This is really helpful for our translators.
